From 755d2667ac03716f88d5207760a493572bc56848 Mon Sep 17 00:00:00 2001
From: Steve MacLean <sdmaclea.qdt@qualcommdatacenter.com>
Date: Fri, 3 Nov 2017 11:20:13 -0400
Subject: [PATCH] [Arm64] SIMD applyCalleeSaveHeuristics

---
 src/jit/lsra.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/jit/lsra.cpp b/src/jit/lsra.cpp
index 6580264..01a6604 100644
--- a/src/jit/lsra.cpp
+++ b/src/jit/lsra.cpp
@@ -721,9 +721,9 @@ void LinearScan::applyCalleeSaveHeuristics(RefPosition* rp)
     Interval* theInterval = rp->getInterval();
 
 #ifdef DEBUG
-    regMaskTP calleeSaveMask = calleeSaveRegs(getRegisterType(theInterval, rp));
     if (doReverseCallerCallee())
     {
+        regMaskTP calleeSaveMask = calleeSaveRegs(theInterval->registerType);
         rp->registerAssignment =
             getConstrainedRegMask(rp->registerAssignment, calleeSaveMask, rp->minRegCandidateCount);
     }
-- 
2.7.4

