From 849e8d5fdf9cabfe6bf60ee8d724276d29ea4347 Mon Sep 17 00:00:00 2001
From: Steve MacLean <sdmaclea.qdt@qualcommdatacenter.com>
Date: Tue, 7 Nov 2017 15:46:31 -0500
Subject: [PATCH] [Arm64] SIMD lvHfaSlots

---
 src/jit/compiler.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/jit/compiler.h b/src/jit/compiler.h
index ffaa8ce..ee2bc4b 100644
--- a/src/jit/compiler.h
+++ b/src/jit/compiler.h
@@ -425,7 +425,7 @@ public:
     unsigned lvHfaSlots() const
     {
         assert(lvIsHfa());
-        assert(lvType == TYP_STRUCT);
+        assert(varTypeIsStruct(lvType));
 #ifdef _TARGET_ARM_
         return lvExactSize / sizeof(float);
 #else  //  _TARGET_ARM64_
-- 
2.7.4

