From 2d9b1d111e62a8a4a4935a6a5cae574133c60066 Mon Sep 17 00:00:00 2001
From: Steve MacLean <sdmaclea.qdt@qualcommdatacenter.com>
Date: Thu, 2 Nov 2017 17:47:30 -0400
Subject: [PATCH] [Arm64] SIMD lvaIsMultiregStruct

---
 src/jit/lclvars.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/jit/lclvars.cpp b/src/jit/lclvars.cpp
index c7f15d8..8ca7fd5 100644
--- a/src/jit/lclvars.cpp
+++ b/src/jit/lclvars.cpp
@@ -2215,7 +2215,11 @@ void Compiler::lvaSetVarDoNotEnregister(unsigned varNum DEBUGARG(DoNotEnregister
 // Returns true if this local var is a multireg struct
 bool Compiler::lvaIsMultiregStruct(LclVarDsc* varDsc)
 {
+#if defined(_TARGET_ARM64_)
+    if (varTypeIsStruct(varDsc->TypeGet()))
+#else
     if (varDsc->TypeGet() == TYP_STRUCT)
+#endif
     {
         CORINFO_CLASS_HANDLE clsHnd = varDsc->lvVerTypeInfo.GetClassHandleForValueClass();
         structPassingKind    howToPassStruct;
-- 
2.7.4

