From 1594a83a74d7beeb39ff764f848c1570b5692735 Mon Sep 17 00:00:00 2001
From: dotnet-maestro-bot <dotnet-maestro-bot@microsoft.com>
Date: Tue, 28 Nov 2017 13:28:56 -0600
Subject: [PATCH] Update CoreClr to preview1-25928-02 (#15240)

---
 ILAsmVersion.txt   | 2 +-
 dependencies.props | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/ILAsmVersion.txt b/ILAsmVersion.txt
index 603c8d1..19ba9ad 100644
--- a/ILAsmVersion.txt
+++ b/ILAsmVersion.txt
@@ -1 +1 @@
-2.1.0-preview1-25927-01
+2.1.0-preview1-25928-02
diff --git a/dependencies.props b/dependencies.props
index d8376cf..0150c85 100644
--- a/dependencies.props
+++ b/dependencies.props
@@ -24,7 +24,7 @@
   <!-- Source of truth for dependency tooling: the commit hash of the dotnet/versions master branch as of the last auto-upgrade. -->
   <PropertyGroup>
     <CoreFxCurrentRef>1ec1f778f861fd1e6386f2f01c8093b2dafa2b1e</CoreFxCurrentRef>
-    <CoreClrCurrentRef>36ae8442ffbd250fbd662edf0f7d9cae10717ab1</CoreClrCurrentRef>
+    <CoreClrCurrentRef>8e2049b7a5c121dca2dceaf274102387e8d465f0</CoreClrCurrentRef>
     <BuildToolsCurrentRef>c4d2b4cae431fc1463a140d671a2fdbe18c2199a</BuildToolsCurrentRef>
     <PgoDataCurrentRef>36ae8442ffbd250fbd662edf0f7d9cae10717ab1</PgoDataCurrentRef>
   </PropertyGroup>
@@ -34,7 +34,7 @@
     <MicrosoftPrivateCoreFxNETCoreAppPackageVersion>4.5.0-preview1-25927-01</MicrosoftPrivateCoreFxNETCoreAppPackageVersion>
     <MicrosoftNETCorePlatformsPackageVersion>2.1.0-preview1-25927-01</MicrosoftNETCorePlatformsPackageVersion>
     <PgoDataPackageVersion>99.99.99-master-20171128-0148</PgoDataPackageVersion>
-    <MicrosoftNETCoreRuntimeCoreCLRPackageVersion>2.1.0-preview1-25927-01</MicrosoftNETCoreRuntimeCoreCLRPackageVersion>
+    <MicrosoftNETCoreRuntimeCoreCLRPackageVersion>2.1.0-preview1-25928-02</MicrosoftNETCoreRuntimeCoreCLRPackageVersion>
     <XunitPackageVersion>2.2.0-beta2-build3300</XunitPackageVersion>
     <XunitConsoleNetcorePackageVersion>1.0.2-prerelease-00177</XunitConsoleNetcorePackageVersion>
     <XunitPerformanceApiPackageVersion>1.0.0-beta-build0012</XunitPerformanceApiPackageVersion>
-- 
2.7.4

