From a66defa47a1f340ab0edcd4765560e2f617e7d92 Mon Sep 17 00:00:00 2001
From: Bruce Forstall <brucefo@microsoft.com>
Date: Tue, 28 Nov 2017 14:31:09 -0800
Subject: [PATCH] Disable muldimjagary test for arm

Fixes #15245
---
 tests/arm/Tests.lst | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/arm/Tests.lst b/tests/arm/Tests.lst
index 042eb83..e2e1400 100644
--- a/tests/arm/Tests.lst
+++ b/tests/arm/Tests.lst
@@ -30497,7 +30497,7 @@ RelativePath=GC\Scenarios\muldimjagary\muldimjagary\muldimjagary.cmd
 WorkingDir=GC\Scenarios\muldimjagary\muldimjagary
 Expected=0
 MaxAllowedDurationSeconds=600
-Categories=EXPECTED_PASS
+Categories=EXPECTED_FAIL;3392
 HostStyle=0
 
 [GCSimulator_261.cmd_3830]
-- 
2.7.4

