From c059a5fb7ef659494956e4e157ee92a2e6ee65e2 Mon Sep 17 00:00:00 2001
From: Bruce Forstall <brucefo@microsoft.com>
Date: Tue, 28 Nov 2017 16:30:06 -0800
Subject: [PATCH] Fix netci.groovy braces after Linux/x86 change

---
 netci.groovy | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/netci.groovy b/netci.groovy
index cb0beed..6909a9b 100755
--- a/netci.groovy
+++ b/netci.groovy
@@ -2167,16 +2167,19 @@ Constants.allScenarios.each { scenario ->
                         }
                     }
                     else {
+                        // Non-Windows
                         if (architecture == 'arm64') {
                             if (scenario != 'default' && scenario != 'r2r' && scenario != 'gcstress0x3' && scenario != 'gcstress0xc') {
                                 return
                             }
+                        }
                         else if (architecture == 'x86') {
                             // Linux/x86 only want default test
                             if (scenario != 'default') {
                                 return
                             }
                         }
+                    }
 
                     // For CentOS, we only want Checked/Release builds.
                     if (os == 'CentOS7.1') {
-- 
2.7.4

