From 6e5e7bfba21db7f5ffd2122ce67a35f45b5e2a42 Mon Sep 17 00:00:00 2001
From: dotnet-maestro-bot <dotnet-maestro-bot@microsoft.com>
Date: Wed, 7 Mar 2018 13:56:51 -0600
Subject: [PATCH] Update CoreFx to preview2-26307-06 (#16798)

---
 dependencies.props | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/dependencies.props b/dependencies.props
index 864707e..2074611 100644
--- a/dependencies.props
+++ b/dependencies.props
@@ -23,7 +23,7 @@
 
   <!-- Source of truth for dependency tooling: the commit hash of the dotnet/versions master branch as of the last auto-upgrade. -->
   <PropertyGroup>
-    <CoreFxCurrentRef>f55c23b14944373897d8fda65b5a9e2ba0a92171</CoreFxCurrentRef>
+    <CoreFxCurrentRef>0dcaa67a086a4e0b505b25d86e92dc0423d220a2</CoreFxCurrentRef>
     <CoreClrCurrentRef>f55c23b14944373897d8fda65b5a9e2ba0a92171</CoreClrCurrentRef>
     <BuildToolsCurrentRef>f55c23b14944373897d8fda65b5a9e2ba0a92171</BuildToolsCurrentRef>
     <PgoDataCurrentRef>553f30357d08df4ed6d32f70f3478a76f5db79c2</PgoDataCurrentRef>
@@ -31,8 +31,8 @@
 
   <!-- Tests/infrastructure dependency versions. -->
   <PropertyGroup>
-    <MicrosoftPrivateCoreFxNETCoreAppPackageVersion>4.5.0-preview2-26307-03</MicrosoftPrivateCoreFxNETCoreAppPackageVersion>
-    <MicrosoftNETCorePlatformsPackageVersion>2.1.0-preview2-26307-03</MicrosoftNETCorePlatformsPackageVersion>
+    <MicrosoftPrivateCoreFxNETCoreAppPackageVersion>4.5.0-preview2-26307-06</MicrosoftPrivateCoreFxNETCoreAppPackageVersion>
+    <MicrosoftNETCorePlatformsPackageVersion>2.1.0-preview2-26307-06</MicrosoftNETCorePlatformsPackageVersion>
     <PgoDataPackageVersion>99.99.99-master-20180228-0037</PgoDataPackageVersion>
     <MicrosoftNETCoreRuntimeCoreCLRPackageVersion>2.1.0-preview2-26307-01</MicrosoftNETCoreRuntimeCoreCLRPackageVersion>
     <XunitPackageVersion>2.2.0-beta2-build3300</XunitPackageVersion>
-- 
2.7.4

