From d5b2f75e9200b459be265839671909bb51d3f8b9 Mon Sep 17 00:00:00 2001
From: Levi Broderick <levib@microsoft.com>
Date: Tue, 6 Mar 2018 09:11:52 -0800
Subject: [PATCH] Fix potential buffer overrun in ClearWithReferences

---
 src/mscorlib/shared/System/SpanHelpers.cs | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/mscorlib/shared/System/SpanHelpers.cs b/src/mscorlib/shared/System/SpanHelpers.cs
index 6752d7f..c3cfe5a 100644
--- a/src/mscorlib/shared/System/SpanHelpers.cs
+++ b/src/mscorlib/shared/System/SpanHelpers.cs
@@ -517,7 +517,6 @@ namespace System
             // Write first four and last three.
             Unsafe.Add(ref ip, 2) = default(IntPtr);
             Unsafe.Add(ref ip, 3) = default(IntPtr);
-            Unsafe.Add(ref ip, 4) = default(IntPtr);
             Unsafe.Add(ref Unsafe.Add(ref ip, (IntPtr)pointerSizeLength), -3) = default(IntPtr);
             Unsafe.Add(ref Unsafe.Add(ref ip, (IntPtr)pointerSizeLength), -2) = default(IntPtr);
 
-- 
2.7.4

