From 5a43bbe82f0eeaef378be99db81005c3d277356f Mon Sep 17 00:00:00 2001
From: Steve MacLean <sdmaclea.qdt@qualcommdatacenter.com>
Date: Mon, 10 Jul 2017 16:27:28 +0000
Subject: [PATCH] PAL sync use FastInterlockOr for Volatile m_dwFlags

---
 src/vm/synch.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/vm/synch.h b/src/vm/synch.h
index d88ec46..b307429 100644
--- a/src/vm/synch.h
+++ b/src/vm/synch.h
@@ -103,21 +103,21 @@ private:
     {
         LIMITED_METHOD_CONTRACT;
         // cannot use `|=' operator on `Volatile<DWORD>'
-        m_dwFlags = m_dwFlags | CLREVENT_FLAGS_AUTO_EVENT;
+        FastInterlockOr(&m_dwFlags, CLREVENT_FLAGS_AUTO_EVENT);
     }
     BOOL IsOSEvent() { LIMITED_METHOD_CONTRACT; return m_dwFlags & CLREVENT_FLAGS_OS_EVENT; }
     void SetOSEvent ()
     {
         LIMITED_METHOD_CONTRACT;
         // cannot use `|=' operator on `Volatile<DWORD>'
-        m_dwFlags = m_dwFlags | CLREVENT_FLAGS_OS_EVENT;
+        FastInterlockOr(&m_dwFlags, CLREVENT_FLAGS_OS_EVENT);
     }
     BOOL IsInDeadlockDetection() { LIMITED_METHOD_CONTRACT; return m_dwFlags & CLREVENT_FLAGS_IN_DEADLOCK_DETECTION; }
     void SetInDeadlockDetection ()
     {
         LIMITED_METHOD_CONTRACT;
         // cannot use `|=' operator on `Volatile<DWORD>'
-        m_dwFlags = m_dwFlags | CLREVENT_FLAGS_IN_DEADLOCK_DETECTION;
+        FastInterlockOr(&m_dwFlags, CLREVENT_FLAGS_IN_DEADLOCK_DETECTION);
     }
 };
 
-- 
2.7.4

