From 008881a0299a32005471735cdcf78ddb826b3cf1 Mon Sep 17 00:00:00 2001
From: Steve MacLean <sdmaclea.qdt@qualcommdatacenter.com>
Date: Wed, 22 Nov 2017 12:04:27 -0500
Subject: [PATCH] Enable GDB JIT features

---
 clrfeatures.cmake | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/clrfeatures.cmake b/clrfeatures.cmake
index 367777a..96fdea2 100644
--- a/clrfeatures.cmake
+++ b/clrfeatures.cmake
@@ -1,3 +1,8 @@
+set(FEATURE_GDBJIT 1)
+set(FEATURE_GDBJIT_FRAME 1)
+set(FEATURE_GDBJIT_LANGID_CS 1)
+set(FEATURE_GDBJIT_SYMTAB 1)
+
 if(CLR_CMAKE_TARGET_TIZEN_LINUX)
   set(FEATURE_GDBJIT_LANGID_CS 1)
 endif()
-- 
2.7.4

