From b3d3ae34dddd3f1b231aa06337905abf8de63cd5 Mon Sep 17 00:00:00 2001
From: Steve MacLean <sdmaclea.qdt@qualcommdatacenter.com>
Date: Thu, 25 Jan 2018 16:57:49 -0500
Subject: [PATCH] [Arm64] Remove Short Vector return workaround

---
 src/jit/importer.cpp | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/src/jit/importer.cpp b/src/jit/importer.cpp
index 378e49b..69c1c6f 100644
--- a/src/jit/importer.cpp
+++ b/src/jit/importer.cpp
@@ -8753,18 +8753,6 @@ REDO_RETURN_NODE:
             return op;
         }
     }
-#if defined(FEATURE_HW_INTRINSICS) && defined(_TARGET_ARM64_)
-    else if ((op->gtOper == GT_HWIntrinsic) && varTypeIsSIMD(op->gtType))
-    {
-        // TODO-ARM64-FIXME Implement ARM64 ABI for Short Vectors properly
-        // assert(op->gtType == info.compRetNativeType)
-        if(op->gtType != info.compRetNativeType)
-        {
-            // Insert a register move to keep target type of SIMD intrinsic intact
-            op = gtNewScalarHWIntrinsicNode(info.compRetNativeType, op, NI_ARM64_NONE_MOV);
-        }
-    }
-#endif
     else if (op->gtOper == GT_COMMA)
     {
         op->gtOp.gtOp2 = impFixupStructReturnType(op->gtOp.gtOp2, retClsHnd);
-- 
2.7.4

