From ec08ca6db1965b11e17f8f47b9a29869198ffe8b Mon Sep 17 00:00:00 2001
From: Steve MacLean <sdmaclea.qdt@qualcommdatacenter.com>
Date: Fri, 20 Oct 2017 16:58:00 -0400
Subject: [PATCH] [Arm64] Fix bogus assert

---
 src/jit/lclvars.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/jit/lclvars.cpp b/src/jit/lclvars.cpp
index e5302ed..ccc226e 100644
--- a/src/jit/lclvars.cpp
+++ b/src/jit/lclvars.cpp
@@ -6150,8 +6150,13 @@ int Compiler::lvaAllocLocalAndSetVirtualOffset(unsigned lclNum, unsigned size, i
 #endif
                             ))
     {
+#ifdef _TARGET_ARM64_
+        // Note that stack offsets are negative or equal to zero
+        assert(stkOffs <= 0);
+#else
         // Note that stack offsets are negative
         assert(stkOffs < 0);
+#endif
 
         // alignment padding
         unsigned pad = 0;
-- 
2.7.4

