From 3aa871de03e613fde7f9fea118dd43c5cbb8282f Mon Sep 17 00:00:00 2001
From: sergey ignatov <sergign60@mail.ru>
Date: Tue, 12 Dec 2017 16:35:19 +0300
Subject: [PATCH] Added comment to explicit number (#15484)

---
 src/inc/gcinfodecoder.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/inc/gcinfodecoder.h b/src/inc/gcinfodecoder.h
index 5151616..fccf67a 100644
--- a/src/inc/gcinfodecoder.h
+++ b/src/inc/gcinfodecoder.h
@@ -240,7 +240,7 @@ public:
         _ASSERTE( pBuffer != NULL );
 
         m_pCurrent = m_pBuffer = dac_cast<PTR_size_t>((size_t)dac_cast<TADDR>(pBuffer) & ~((size_t)sizeof(size_t)-1));
-        m_RelPos = m_InitialRelPos = (int)((size_t)dac_cast<TADDR>(pBuffer) % sizeof(size_t)) * 8;
+        m_RelPos = m_InitialRelPos = (int)((size_t)dac_cast<TADDR>(pBuffer) % sizeof(size_t)) * 8/*BITS_PER_BYTE*/;
     }
 
     BitStreamReader(const BitStreamReader& other)
-- 
2.7.4

