From 739b71daf0249526ff359d5809c48e2ce38a07ac Mon Sep 17 00:00:00 2001
From: Egor Chesakov <Egor.Chesakov@microsoft.com>
Date: Tue, 12 Dec 2017 06:30:33 -0800
Subject: [PATCH] Remove undefined and unused getPersonalityValue function from
 CEECompileInfo (#15482)

---
 src/vm/compile.h | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/vm/compile.h b/src/vm/compile.h
index 5b932d5..94c3ebc 100644
--- a/src/vm/compile.h
+++ b/src/vm/compile.h
@@ -351,10 +351,6 @@ class CEECompileInfo : public ICorCompileInfo
             IN  CORINFO_METHOD_HANDLE    hMethod,
             OUT CORJIT_FLAGS            *pFlags);
 
-#ifdef _WIN64
-    SIZE_T  getPersonalityValue();
-#endif
-
     void* GetStubSize(void *pStubAddress, DWORD *pSizeToCopy);
 
     HRESULT GetStubClone(void *pStub, BYTE *pBuffer, DWORD dwBufferSize);
-- 
2.7.4

