From 168ff9b8565f0fdaef02c08b8f6b690da86a340d Mon Sep 17 00:00:00 2001
From: dotnet-maestro-bot <dotnet-maestro-bot@microsoft.com>
Date: Tue, 11 Jul 2017 13:45:29 +0000
Subject: [PATCH] Update CoreClr, CoreFx to preview2-25510-01,
 preview2-25511-02, respectively

---
 dependencies.props | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/dependencies.props b/dependencies.props
index 98fdd15..389df6c 100644
--- a/dependencies.props
+++ b/dependencies.props
@@ -13,19 +13,19 @@
   
   <!-- Source of truth for dependency tooling: the commit hash of the dotnet/versions master branch as of the last auto-upgrade. -->
   <PropertyGroup>
-    <CoreFxCurrentRef>d6c839cebfc879fa6b7d0d953b3788c9d19ad5f0</CoreFxCurrentRef>
-    <CoreClrCurrentRef>d6c839cebfc879fa6b7d0d953b3788c9d19ad5f0</CoreClrCurrentRef>
+    <CoreFxCurrentRef>e7e2bd1f8c78028e2a188dc15d758d4c4624989e</CoreFxCurrentRef>
+    <CoreClrCurrentRef>e7e2bd1f8c78028e2a188dc15d758d4c4624989e</CoreClrCurrentRef>
   </PropertyGroup>
 
   <!-- Auto-upgraded properties for each build info dependency. -->
   <PropertyGroup>
-    <CoreFxPackageVersion>4.5.0-preview2-25508-04</CoreFxPackageVersion>
+    <CoreFxPackageVersion>4.5.0-preview2-25511-02</CoreFxPackageVersion>
     <PlatformPackageVersion>2.1.0-preview1-25324-02</PlatformPackageVersion>
   </PropertyGroup>
 
   <!-- Full package version strings that are used in other parts of the build. -->
   <PropertyGroup>
-    <CoreClrPackageVersion>2.1.0-preview2-25508-04</CoreClrPackageVersion>
+    <CoreClrPackageVersion>2.1.0-preview2-25510-01</CoreClrPackageVersion>
     <XunitPackageVersion>2.2.0-beta2-build3300</XunitPackageVersion>
     <XUnitConsoleNetCoreVersion>1.0.2-prerelease-00177</XUnitConsoleNetCoreVersion>
     <XUnitPerformanceApiVersion>1.0.0-beta-build0007</XUnitPerformanceApiVersion>
-- 
2.7.4

