From 47b11d513a347c8a8e6ea3ceca2a0d78cacf52fe Mon Sep 17 00:00:00 2001
From: Steve MacLean <sdmaclea.qdt@qualcommdatacenter.com>
Date: Fri, 3 Nov 2017 14:51:08 -0400
Subject: [PATCH] [Arm64] emitIns_R_S imm size

---
 src/jit/emitarm64.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/jit/emitarm64.cpp b/src/jit/emitarm64.cpp
index db31d7d..64d9b6f 100644
--- a/src/jit/emitarm64.cpp
+++ b/src/jit/emitarm64.cpp
@@ -6320,7 +6320,7 @@ void emitter::emitIns_R_S(instruction ins, emitAttr attr, regNumber reg1, int va
         if (useRegForImm)
         {
             regNumber rsvdReg = codeGen->rsGetRsvdReg();
-            codeGen->instGen_Set_Reg_To_Imm(size, rsvdReg, imm);
+            codeGen->instGen_Set_Reg_To_Imm(EA_8BYTE, rsvdReg, imm);
             fmt = IF_LS_3A;
         }
     }
-- 
2.7.4

