From 2d12c9a1f5692a85ff2044d9ee97a90494d4edae Mon Sep 17 00:00:00 2001
From: Steve MacLean <sdmaclea.qdt@qualcommdatacenter.com>
Date: Wed, 20 Sep 2017 18:04:36 -0400
Subject: [PATCH] [Arm64] Implement JCC/SETCC nodes

---
 src/jit/codegenarmarch.cpp | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/src/jit/codegenarmarch.cpp b/src/jit/codegenarmarch.cpp
index a41fe96..7692c7c 100644
--- a/src/jit/codegenarmarch.cpp
+++ b/src/jit/codegenarmarch.cpp
@@ -246,8 +246,6 @@ void CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
             genCodeForJumpTrue(treeNode);
             break;
 
-#ifdef _TARGET_ARM_
-
         case GT_JCC:
             genCodeForJcc(treeNode->AsCC());
             break;
@@ -256,8 +254,6 @@ void CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
             genCodeForSetcc(treeNode->AsCC());
             break;
 
-#endif // _TARGET_ARM_
-
         case GT_RETURNTRAP:
             genCodeForReturnTrap(treeNode->AsOp());
             break;
@@ -3284,7 +3280,6 @@ void CodeGen::genCodeForJumpTrue(GenTreePtr tree)
     }
 }
 
-#if defined(_TARGET_ARM_)
 
 //------------------------------------------------------------------------
 // genCodeForJcc: Produce code for a GT_JCC node.
@@ -3325,6 +3320,9 @@ void CodeGen::genCodeForSetcc(GenTreeCC* setcc)
     // Make sure nobody is setting GTF_RELOP_NAN_UN on this node as it is ignored.
     assert((setcc->gtFlags & GTF_RELOP_NAN_UN) == 0);
 
+#ifdef _TARGET_ARM64_
+    inst_SET(jumpKind, dstReg);
+#else
     // Emit code like that:
     //   ...
     //   bgt True
@@ -3346,11 +3344,11 @@ void CodeGen::genCodeForSetcc(GenTreeCC* setcc)
     genDefineTempLabel(labelTrue);
     getEmitter()->emitIns_R_I(INS_mov, emitActualTypeSize(setcc->TypeGet()), dstReg, 1);
     genDefineTempLabel(labelNext);
+#endif
 
     genProduceReg(setcc);
 }
 
-#endif // defined(_TARGET_ARM_)
 
 //------------------------------------------------------------------------
 // genCodeForStoreBlk: Produce code for a GT_STORE_OBJ/GT_STORE_DYN_BLK/GT_STORE_BLK node.
-- 
2.7.4

