From bd96214b4297bfacb0f7e1b6f75fea1face641bf Mon Sep 17 00:00:00 2001
From: Steve MacLean <sdmaclea.qdt@qualcommdatacenter.com>
Date: Tue, 31 Oct 2017 16:05:33 -0400
Subject: [PATCH] [Arm64] Fix SIMD ContainCheckRet()

---
 src/jit/lower.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/jit/lower.cpp b/src/jit/lower.cpp
index f62efd1..c044c4e 100644
--- a/src/jit/lower.cpp
+++ b/src/jit/lower.cpp
@@ -5819,6 +5819,12 @@ void Lowering::ContainCheckRet(GenTreeOp* ret)
             {
                 MakeSrcContained(ret, op1);
             }
+#if defined(_TARGET_ARM64_) && defined(FEATURE_SIMD)
+            if (varTypeIsSIMD(op1))
+            {
+                MakeSrcContained(ret, op1);
+            }
+#endif // defined(_TARGET_ARM64_) && defined(FEATURE_SIMD)
         }
     }
 #endif // FEATURE_MULTIREG_RET
-- 
2.7.4

