From a6ab1d3dea256cd84c3cfa02ac389531596af16a Mon Sep 17 00:00:00 2001
From: Steve MacLean <sdmaclea.qdt@qualcommdatacenter.com>
Date: Tue, 28 Nov 2017 19:46:43 -0500
Subject: [PATCH] [Arm64] Clear contained on SIMDIntrinsicGetItem op1

---
 src/jit/lowerarmarch.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/jit/lowerarmarch.cpp b/src/jit/lowerarmarch.cpp
index 51d4a3b..8ebf224 100644
--- a/src/jit/lowerarmarch.cpp
+++ b/src/jit/lowerarmarch.cpp
@@ -785,14 +785,17 @@ void Lowering::ContainCheckSIMD(GenTreeSIMD* simdNode)
 
         case SIMDIntrinsicGetItem:
         {
-            // TODO-ARM64-CQ Support containing op1 memory ops
 
             // This implements get_Item method. The sources are:
             //  - the source SIMD struct
             //  - index (which element to get)
             // The result is baseType of SIMD struct.
+            op1 = simdNode->gtOp.gtOp1;
             op2 = simdNode->gtOp.gtOp2;
 
+            // TODO-ARM64-CQ Support containing op1 memory ops
+            op1->ClearContained();
+
             // If the index is a constant, mark it as contained.
             if (op2->IsCnsIntOrI())
             {
-- 
2.7.4

