From 90f1918d672eec9cea327c20a7a40ac96b318af1 Mon Sep 17 00:00:00 2001
From: Jacek Blaszczynski <biosciencenow@outlook.com>
Date: Thu, 11 Jan 2018 13:29:26 +0100
Subject: [PATCH] movhlps categorized as dstdstsrc

---
 src/jit/emitxarch.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/jit/emitxarch.cpp b/src/jit/emitxarch.cpp
index 9c9bff1..80da26d 100644
--- a/src/jit/emitxarch.cpp
+++ b/src/jit/emitxarch.cpp
@@ -105,6 +105,7 @@ bool emitter::IsDstDstSrcAVXInstruction(instruction ins)
         case INS_minps:
         case INS_minsd:
         case INS_minss:
+        case INS_movlhps:
         case INS_mulpd:
         case INS_mulps:
         case INS_mulsd:
@@ -189,7 +190,6 @@ bool emitter::IsDstSrcSrcAVXInstruction(instruction ins)
         case INS_cvtss2sd:
         case INS_movhpd:
         case INS_movhps:
-        case INS_movlhps:
         case INS_movlpd:
         case INS_movlps:
         case INS_movss:
-- 
2.7.4

