From e0c33c7e8e023cf581b8816cad1b85496d49c7b3 Mon Sep 17 00:00:00 2001
From: Davis Goodin <dagood@users.noreply.github.com>
Date: Wed, 22 Nov 2017 09:05:51 -0600
Subject: [PATCH] Update BuildTools to 2.0.0-servicing-02221-03 (#295)

---
 BuildToolsVersion.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/BuildToolsVersion.txt b/BuildToolsVersion.txt
index 5e40414..67ce4c9 100644
--- a/BuildToolsVersion.txt
+++ b/BuildToolsVersion.txt
@@ -1 +1 @@
-2.0.0-servicing-02217-01
+2.0.0-servicing-02221-03
-- 
2.7.4

