From 319fbdc0e101ea4dc1fd5610e716a7475286c8e2 Mon Sep 17 00:00:00 2001
From: Tom Deseyn <tom.deseyn@gmail.com>
Date: Fri, 24 Nov 2017 09:47:29 +0100
Subject: [PATCH] Update tarball commit info to include git describe

---
 build-source-tarball.sh | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/build-source-tarball.sh b/build-source-tarball.sh
index 878566c..fb1ecac 100755
--- a/build-source-tarball.sh
+++ b/build-source-tarball.sh
@@ -69,7 +69,10 @@ do
 done
 
 # Record commits for the source-build repo and all submodules, to aid in reproducibility.
-echo -e "path\tchecked-in\tactual" > $TARBALL_ROOT/commits.txt
-echo -e "source-build\t$(git rev-parse HEAD)\t$(git rev-parse HEAD)" >> $TARBALL_ROOT/commits.txt
-git submodule foreach --quiet --recursive "actual=\$(git rev-parse HEAD); echo -e \"\$toplevel/\$path\t\$sha1\t\$actual\" >> $FULL_TARBALL_ROOT/commits.txt"
+cat >$TARBALL_ROOT/source-build-info.txt << EOF
+source-build:
+ $(git rev-parse HEAD) . ($(git describe --always HEAD))
 
+submodules:
+$(git submodule status)
+EOF
-- 
2.7.4

