From 022c4992cc140a698d2d96a19da9164c7984e6b3 Mon Sep 17 00:00:00 2001
From: Dan Seefeldt <Dan.Seefeldt@microsoft.com>
Date: Fri, 9 Feb 2018 08:22:52 -0600
Subject: [PATCH] Temporarily diable the ARM build (#325)

---
 netci.groovy | 42 ------------------------------------------
 1 file changed, 42 deletions(-)

diff --git a/netci.groovy b/netci.groovy
index 8678150..49d3d02 100644
--- a/netci.groovy
+++ b/netci.groovy
@@ -99,45 +99,3 @@ def addPushJob(String project, String branch, String os, String configuration)
   }
 }
 
-[true, false].each { isPR ->
-  ["Linux_ARM"].each { os->
-    ["Release", "Debug"].each { configuration ->
-
-      def shortJobName = "${os}_${configuration}";
-      def contextString = "${os} ${configuration}";
-      def triggerPhrase = "(?i).*test\\W+${contextString}.*";
-
-      def newJob = job(Utilities.getFullJobName(project, shortJobName, isPR)){
-        steps{
-            shell("git submodule update --init --recursive");
-            shell("./init-tools.sh")
-            if(os == "Linux_ARM"){
-              shell("./arm-ci.sh arm ./build.sh /p:Platform=arm /p:Configuration=${configuration} ${loggingOptions}")
-            }
-            if(os == "Tizen"){
-              shell("./arm-ci.sh armel ./build.sh /p:Platform=armel /p:Configuration=${configuration} ${loggingOptions}")
-            }
-        }
-      }
-
-      Utilities.setMachineAffinity(newJob, 'Ubuntu', 'arm-cross-latest');
-
-      Utilities.standardJobSetup(newJob, project, isPR, "*/${branch}");
-      if(isPR){
-        //We run Tizen Release and Ubuntu ARM Release
-        if(configuration == "Release"){
-          Utilities.addGithubPRTriggerForBranch(newJob, branch, contextString);
-        }
-        else{
-          Utilities.addGithubPRTriggerForBranch(newJob, branch, contextString, triggerPhrase);
-        }
-      }
-      else{
-        Utilities.addGithubPushTrigger(newJob);
-      }
-
-    }
-  }
-}
-
-
-- 
2.7.4

