From 6514bbb34fa9a235552532b23a19ae4990eb8e87 Mon Sep 17 00:00:00 2001
From: Steve MacLean <sdmaclea.qdt@qualcommdatacenter.com>
Date: Mon, 12 Jun 2017 20:52:46 +0000
Subject: [PATCH] [Arm64/Unix] Implement YieldProcessor()

---
 src/pal/inc/pal.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/pal/inc/pal.h b/src/pal/inc/pal.h
index 2d703e8..499b633 100644
--- a/src/pal/inc/pal.h
+++ b/src/pal/inc/pal.h
@@ -4572,6 +4572,8 @@ YieldProcessor(
     __asm__ __volatile__(
         "rep\n"
         "nop");
+#elif defined(_ARM64_)
+    __asm__ __volatile__( "yield");
 #else
     return;
 #endif
-- 
2.7.4

