From 1e6ed5ba901a4cf4f5293bfe0f44f409c2be372b Mon Sep 17 00:00:00 2001
From: Steve MacLean <sdmaclea.qdt@qualcommdatacenter.com>
Date: Tue, 28 Nov 2017 13:20:55 -0500
Subject: [PATCH] [Arm64] Enable SIMD in crossgen

---
 src/zap/zapper.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/zap/zapper.cpp b/src/zap/zapper.cpp
index 3765135..7bf1ead 100644
--- a/src/zap/zapper.cpp
+++ b/src/zap/zapper.cpp
@@ -1394,6 +1394,14 @@ void Zapper::InitializeCompilerFlags(CORCOMPILE_VERSION_INFO * pVersionInfo)
 
 #endif // _TARGET_X86_
 
+#if defined(_TARGET_ARM64_)
+    static ConfigDWORD fFeatureSIMD;
+    if (fFeatureSIMD.val(CLRConfig::EXTERNAL_FeatureSIMD) != 0)
+    {
+        m_pOpt->m_compilerFlags.Set(CORJIT_FLAGS::CORJIT_FLAG_FEATURE_SIMD);
+    }
+#endif
+
     if (   m_pOpt->m_compilerFlags.IsSet(CORJIT_FLAGS::CORJIT_FLAG_DEBUG_INFO)
         && m_pOpt->m_compilerFlags.IsSet(CORJIT_FLAGS::CORJIT_FLAG_DEBUG_CODE)
         && m_pOpt->m_compilerFlags.IsSet(CORJIT_FLAGS::CORJIT_FLAG_PROF_ENTERLEAVE))
-- 
2.7.4

