From 8d5696535482cd34ee56974168f9831611d15945 Mon Sep 17 00:00:00 2001
From: Steve MacLean <sdmaclea.qdt@qualcommdatacenter.com>
Date: Fri, 21 Jul 2017 21:44:31 -0400
Subject: [PATCH] [Arm64] Fix ResolveStub tokenHash lookup

---
 src/vm/arm64/virtualcallstubcpu.hpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/vm/arm64/virtualcallstubcpu.hpp b/src/vm/arm64/virtualcallstubcpu.hpp
index b7c5209..68c9125 100644
--- a/src/vm/arm64/virtualcallstubcpu.hpp
+++ b/src/vm/arm64/virtualcallstubcpu.hpp
@@ -217,8 +217,8 @@ struct ResolveHolder
          //w13- this._hashedToken
          //ldr w13, [x10 + DATA_OFFSET(_hashedToken)]
          offset = DATA_OFFSET(_hashedToken);
-         _ASSERTE(offset >=0 && offset%8 == 0);
-         _stub._resolveEntryPoint[n++] = 0xB940014D | offset<<7;
+         _ASSERTE(offset >=0 && offset%4 == 0);
+         _stub._resolveEntryPoint[n++] = 0xB940014D | offset<<8;
 
          //eor x9,x9,x13
          _stub._resolveEntryPoint[n++] = 0xCA0D0129;
-- 
2.7.4

